File: backend/app/api/v1/endpoints/lesson_plan.py
"""Lesson-plan endpoint for retrieving the learner's next speaking drill."""

from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.db.database import get_db
from app.schemas import LessonPlanResponse
from app.services.lesson_service import get_latest_or_default_lesson_plan

router = APIRouter(tags=["lesson-plan"])


@router.get("/lesson-plan/{user_id}", response_model=LessonPlanResponse)
def get_lesson_plan(user_id: str, db: Session = Depends(get_db)) -> LessonPlanResponse:
    """Return the latest lesson plan, or a starter plan for a new learner."""
    return get_latest_or_default_lesson_plan(db, user_id=user_id)

File: frontend/lib/api/lessonPlan.ts
import { apiFetch } from "./client";
import type { LessonPlanResponse } from "@/types/api";

export async function getLessonPlan(userId: string): Promise<LessonPlanResponse> {
  return apiFetch<LessonPlanResponse>(`/lesson-plan/${encodeURIComponent(userId)}`);
}
